PUPIL-1352 feat: added A/B variants of downloads page that render based on posth…

charsville opened the pull request:
## Description

Music year: 1963

Uses a feature flag in posthog to change the layout of the lesson downloads page. The 2 versions are:
A = Original 
B = Downloads are grouped and right side form is sticky

## Issue(s)
PUPIL-1347
PUPIL-1352

## How to test

1. Go to https://deploy-preview-3436--oak-web-application.netlify.thenational.academy/teachers/programmes/computing-primary-ks2/units/photo-editing/lessons/cloning-digital-images/downloads?preselected=all
2. Toggle the feature flags with the posthog toolbar
3. You should see the original variant, and also the new variant from the designs

## Things to note
- [ ] The lesson share page was sharing the component, I have managed to avoid any changes even with the shared component. So the share page should have NOT changed.
- [ ] The ResourceCard component on the downloads page is also used on the curriculum downloads page and the curriculum resource selector component. We should check for no regression/changes there.
- [ ] The new variant has made the right side form sticky, this should be tested on different screen sizes
- [ ] We should check all of the various states on the download page, e.g logged in or not, download confirmation, loading states, no downloads available and risk assessment banner

## Grouping test
As part of this work we needed to group all of the different file types. This is the list of file types in the schema:
```
presentation
intro-quiz-questions
intro-quiz-answers
exit-quiz-questions
exit-quiz-answers
worksheet-pdf
worksheet-pptx
supplementary-pdf
supplementary-docx
curriculum-pdf
lesson-guide-pdf
additional-files
```
I am not 100% sure what all of these files actually are, and so I have done my best guess at grouping this (with help from Mike). The conclusion we have currently come to is that the based on the headings in the designs, this is how they are grouped:

**Slide deck**
`presentation`

**Quizzes**
`intro-quiz-questions`
`intro-quiz-answers`
`exit-quiz-questions`
`exit-quiz-answers`

**Worksheet**
`worksheet-pdf`
`worksheet-pptx`

**Lesson files**
`supplementary-pdf`
`supplementary-docx`
`curriculum-pdf`
`lesson-guide-pdf`

**Additional material**
`additional-files`

We may need to move some types around into different groups @RBrownsett you will probably know more about what types of files can be downloaded and the group they should belong to.

## Designs

[Link to designs](https://www.figma.com/design/3i0y2PHT6pECaGgvLhoN8z/Lesson-page?node-id=4299-47160&m=dev)

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change
- [ ] Check for share page regressions
- [ ] Check curriculum downloads for regression
- [ ] Check curriculum resource selector component for regression
- [ ] New variant sticky form section is responsive
- [ ] Check downloads page states - logged in/out, download confirm, loading state, error state, no downloads, risk assessment banner
- [ ] Check grouping of file types is correct

charsville commented:
> @charsville just one thing before I QA - noticed in the percy tests the size of the existing download card components has been reduced (back to how they used to look). The reason Weronika made them bigger was because some of the downloadable files has long names e.g. slug `lighting-and-cameras` ![image](https://private-user-images.githubusercontent.com/30384846/447058915-407fd6fd-735a-4e7c-abfb-60dafe3afcd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTE3ODcsIm5iZiI6MTc0ODAxMTQ4NywicGF0aCI6Ii8zMDM4NDg0Ni80NDcwNTg5MTUtNDA3ZmQ2ZmQtNzM1YS00ZTdjLWFiZmItNjBkYWZlM2FmY2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4OTE3MjA0MzgwY2Y4NmFkZjNjYTQzY2RlYTFiZTZiNzU2Y2VmZTllNzhiMDRkNDlkYjdkOGEzNjg5YjY3MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DZCoY_AzTp0lXZ-ksIUT9TRjabWHDK4rVSXV49P7i58)
> 
> Not a huge issues (not many have long titles) but wondering if there is another way around it?
> 
> Current live ![image](https://private-user-images.githubusercontent.com/30384846/447059704-bf8aae0b-a568-4404-b2e2-e03d5dcaa642.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTE3ODcsIm5iZiI6MTc0ODAxMTQ4NywicGF0aCI6Ii8zMDM4NDg0Ni80NDcwNTk3MDQtYmY4YWFlMGItYTU2OC00NDA0LWIyZTItZTAzZDVkY2FhNjQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmMmVjZGY2NTIxMWRiYTJkNmQ2MzI3YjRlZmEzYWE4ZmM2YmQ4YzRmYzA1MmJlOTJjNTE5OGEwM2M4MjgwNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FCSm6QkTEe8iwxrlxc75LmBbPz__vhpkS6WJTD81f80)

Ahhh I see, probably one for @messmike to answer. One option would be to have ellipsis for the text that overflows beyond 1 line like this screenshot

![image](https://github.com/user-attachments/assets/2d4e88d5-c05c-45b3-a711-ab71cb38317a)

RBrownsett commented:
> > @charsville just one thing before I QA - noticed in the percy tests the size of the existing download card components has been reduced (back to how they used to look). The reason Weronika made them bigger was because some of the downloadable files has long names e.g. slug `lighting-and-cameras` ![image](https://private-user-images.githubusercontent.com/30384846/447058915-407fd6fd-735a-4e7c-abfb-60dafe3afcd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTE3ODcsIm5iZiI6MTc0ODAxMTQ4NywicGF0aCI6Ii8zMDM4NDg0Ni80NDcwNTg5MTUtNDA3ZmQ2ZmQtNzM1YS00ZTdjLWFiZmItNjBkYWZlM2FmY2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4OTE3MjA0MzgwY2Y4NmFkZjNjYTQzY2RlYTFiZTZiNzU2Y2VmZTllNzhiMDRkNDlkYjdkOGEzNjg5YjY3MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DZCoY_AzTp0lXZ-ksIUT9TRjabWHDK4rVSXV49P7i58)
> > Not a huge issues (not many have long titles) but wondering if there is another way around it?
> > Current live ![image](https://private-user-images.githubusercontent.com/30384846/447059704-bf8aae0b-a568-4404-b2e2-e03d5dcaa642.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTE3ODcsIm5iZiI6MTc0ODAxMTQ4NywicGF0aCI6Ii8zMDM4NDg0Ni80NDcwNTk3MDQtYmY4YWFlMGItYTU2OC00NDA0LWIyZTItZTAzZDVkY2FhNjQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmMmVjZGY2NTIxMWRiYTJkNmQ2MzI3YjRlZmEzYWE4ZmM2YmQ4YzRmYzA1MmJlOTJjNTE5OGEwM2M4MjgwNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FCSm6QkTEe8iwxrlxc75LmBbPz__vhpkS6WJTD81f80)
> 
> Ahhh I see, probably one for @messmike to answer. One option would be to have ellipsis for the text that overflows beyond 1 line like this screenshot
> 
> ![image](https://private-user-images.githubusercontent.com/16628608/447062432-2d4e88d5-c05c-45b3-a711-ab71cb38317a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTIwMTgsIm5iZiI6MTc0ODAxMTcxOCwicGF0aCI6Ii8xNjYyODYwOC80NDcwNjI0MzItMmQ0ZTg4ZDUtYzA1Yy00NWIzLWE3MTEtYWI3MWNiMzgzMTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDgzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxYzI2YTQ1NTI2ZmI0MGM1MmU2NWMxMmEwOTVlMmUxYTAwOGI4MjIzMGVkOTBlYTcwODJhMzUzMjEwMzQ5Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7-aXQT-JlR13XwfMDxP3l5SNEVXK_pjn_fOLTPkTrkI)

Yep that could work. @messmike what do you think?

messmike commented:
> > > @charsville just one thing before I QA - noticed in the percy tests the size of the existing download card components has been reduced (back to how they used to look). The reason Weronika made them bigger was because some of the downloadable files has long names e.g. slug `lighting-and-cameras` ![image](https://private-user-images.githubusercontent.com/30384846/447058915-407fd6fd-735a-4e7c-abfb-60dafe3afcd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTE3ODcsIm5iZiI6MTc0ODAxMTQ4NywicGF0aCI6Ii8zMDM4NDg0Ni80NDcwNTg5MTUtNDA3ZmQ2ZmQtNzM1YS00ZTdjLWFiZmItNjBkYWZlM2FmY2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4OTE3MjA0MzgwY2Y4NmFkZjNjYTQzY2RlYTFiZTZiNzU2Y2VmZTllNzhiMDRkNDlkYjdkOGEzNjg5YjY3MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DZCoY_AzTp0lXZ-ksIUT9TRjabWHDK4rVSXV49P7i58)
> > > Not a huge issues (not many have long titles) but wondering if there is another way around it?
> > > Current live ![image](https://private-user-images.githubusercontent.com/30384846/447059704-bf8aae0b-a568-4404-b2e2-e03d5dcaa642.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTE3ODcsIm5iZiI6MTc0ODAxMTQ4NywicGF0aCI6Ii8zMDM4NDg0Ni80NDcwNTk3MDQtYmY4YWFlMGItYTU2OC00NDA0LWIyZTItZTAzZDVkY2FhNjQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmMmVjZGY2NTIxMWRiYTJkNmQ2MzI3YjRlZmEzYWE4ZmM2YmQ4YzRmYzA1MmJlOTJjNTE5OGEwM2M4MjgwNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FCSm6QkTEe8iwxrlxc75LmBbPz__vhpkS6WJTD81f80)
> > 
> > 
> > Ahhh I see, probably one for @messmike to answer. One option would be to have ellipsis for the text that overflows beyond 1 line like this screenshot
> > ![image](https://private-user-images.githubusercontent.com/16628608/447062432-2d4e88d5-c05c-45b3-a711-ab71cb38317a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTIwMTgsIm5iZiI6MTc0ODAxMTcxOCwicGF0aCI6Ii8xNjYyODYwOC80NDcwNjI0MzItMmQ0ZTg4ZDUtYzA1Yy00NWIzLWE3MTEtYWI3MWNiMzgzMTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIzVDE0NDgzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxYzI2YTQ1NTI2ZmI0MGM1MmU2NWMxMmEwOTVlMmUxYTAwOGI4MjIzMGVkOTBlYTcwODJhMzUzMjEwMzQ5Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7-aXQT-JlR13XwfMDxP3l5SNEVXK_pjn_fOLTPkTrkI)
> 
> Yep that could work. @messmike what do you think?

Yeah, I think truncating it is probably the best option. Thanks both!

RBrownsett commented:
@charsville @messmike Lesson Guides currently show in the Lesson Slides section - I think this would better in its own area (Lesson guides) example slug: `passing-and-receiving-in-games`
![image](https://github.com/user-attachments/assets/14b07d21-d71f-473c-9ca9-0e93fc937440)

RBrownsett commented:
One note on accessibility - is it worth putting each of the sections into their own fieldset so its clear to screen reader users that we have separated them out? E.g

```
  <fieldset>
    <legend>Slide deck</legend>
    <label><input type="checkbox" name="resources" value="presentation"> Slide deck (PPTX)</label>
  </fieldset>

  <fieldset>
    <legend>Quizzes</legend>
    <label><input type="checkbox" name="resources" value="intro-quiz-questions"> Starter quiz questions</label>
    <label><input type="checkbox" name="resources" value="intro-quiz-answers"> Starter quiz answers</label>
    <label><input type="checkbox" name="resources" value="exit-quiz-questions"> Exit quiz questions</label>
    <label><input type="checkbox" name="resources" value="exit-quiz-answers"> Exit quiz answers</label>
  </fieldset>.
```

charsville commented:
@RBrownsett @messmike updated it with these changes:
1. Fixed spacing between groups
2. Added fieldset, legends and labels for accessibility
3. Broke out the lesson guide from "Slide deck" into new group labeled "Lesson guide"
